myanimelist/media.py

```diff
@@ -270,8 +270,11 @@ def parse_sidebar(self, media_page):
             if len(rank_tag_results) == 0:
                 raise Exception("Couldn't find rank tag.")
             # rank_tag is a lxml.etree._ElementUnicodeResult here:
-            rank_tag = rank_tag_results[0].getparent().xpath(".//text()[contains(.,'#')]")[0]
-            media_info['rank'] = int(rank_tag.strip()[1:].replace(',', ''))
+            try:
+                rank_tag = rank_tag_results[0].getparent().xpath(".//text()[contains(.,'#')]")[0]
+                media_info['rank'] = int(rank_tag.strip()[1:].replace(',', ''))
+            except IndexError: # It may happen
```

Comment:
It would be better to check if the returned list contains any items instead of just catching an `IndexError`.